paddle/phi/kernels/onednn/interpolate_kernel.cc

```diff
@@ -15,11 +15,41 @@
 #include "paddle/phi/kernels/interpolate_kernel.h"
 
 #include "paddle/phi/backends/onednn/onednn_reuse.h"
+#include "paddle/phi/core/compat/get_kerneltype_forvar_utils.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/interpolate_function.h"
 
 namespace phi {
 
+KernelKey InterpolateGetKernelTypeForVar(
+    const GetKernelTypeForVarContext* ctx) {
+  const std::string& var_name = ctx->GetVarName();
+  const DenseTensor& tensor = ctx->GetTensor();
+  const KernelKey& expected_kernel_type = ctx->GetKernelKey();
+  const AttributeMap& attrs = ctx->GetAttrs();
+  // Only input require reshaping, weights and
+  // bias are having shape in NCHW order
+  if ((expected_kernel_type.layout() == DataLayout::ONEDNN) &&
+      (tensor.layout() != DataLayout::ONEDNN)) {
+    auto it = attrs.find("data_layout");
```

Comment:
Hi, I'm a little confusing about the definition of `data_layout` and `data_format` in Paddle. In [LRNOp::GetKernelTypeForVar](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/lrn_op.cc#L240), we get the `data_format` from Attrs, but here we get the `data_layout` instead. Do they have any difference? 

 BTW, do we also plan to move the GetKernelTypeForVar of LRNOp/Pad2dOp/Pad3dOp to phi? since there are also some MKLDNN specific code.

Reply:
The Attrs name `data_format`and `data_layout` have the same meaning. However we can't change them for compatibility. 
We also plan to move the GetKernelTypeForVar of LRNOp/Pad2dOp/Pad3dOp to phi soon.